src/System.DirectoryServices.AccountManagement/tests/AccountManagementTests.cs

```diff
@@ -0,0 +1,510 @@
+                Assert.NotNull(p);
+                Assert.Equal(userName, p.Name);
+                Assert.Equal(firstName, p.GivenName);
+                Assert.Equal(lastName, p.Surname);
+                Assert.Equal(displayName, p.DisplayName);
+                Assert.True(p.DistinguishedName.IndexOf(userName, StringComparison.OrdinalIgnoreCase) >= 0);
+                Assert.Equal(userName, p.SamAccountName);
+            }
+        }
+
+        private void ValidateRecentAddedGroup(PrincipalContext context, string groupName, string description, string displayName)
+        {
+            using (GroupPrincipal p = FindGroup(groupName, context))
+            {
+                Assert.NotNull(p);
+                Assert.Equal(groupName, p.Name);
+                Assert.Equal(description, p.Description);
+                Assert.Equal(displayName, p.DisplayName);
+                Assert.Equal(groupName, p.SamAccountName);
+                Assert.True(p.DistinguishedName.IndexOf(groupName, StringComparison.OrdinalIgnoreCase) >= 0);
+            }
+        }
+
+        private void ValidateUserUsingPrincipal(PrincipalContext context, UserPrincipal user)
+        {
+            using (UserPrincipal p = FindUser(user.Name, context))
+            {
+                Assert.NotNull(p);
+                Assert.Equal(user.Name, p.Name);
+                Assert.Equal(user.GivenName, p.GivenName);
+                Assert.Equal(user.Surname, p.Surname);
+                Assert.Equal(user.DisplayName, p.DisplayName);
+                Assert.Equal(user.SamAccountName, p.SamAccountName);
+
+                Assert.Equal(user.Guid, p.Guid);
+                Assert.Equal(user.Sid, p.Sid);
+                Assert.Equal(user.UserPrincipalName, p.UserPrincipalName);
+                Assert.Equal(user.UserCannotChangePassword, p.UserCannotChangePassword);
+                Assert.Equal(user.Enabled, p.Enabled);
+                Assert.Equal(user.AccountExpirationDate, p.AccountExpirationDate);
+            }
+        }
+
+        private void ValidateGroupUsingPrincipal(PrincipalContext context, GroupPrincipal group)
+        {
+            using (GroupPrincipal p = FindGroup(group.Name, context))
+            {
+                Assert.NotNull(p);
+                Assert.Equal(group.Name, p.Name);
+                Assert.Equal(group.DisplayName, p.DisplayName);
+                Assert.Equal(group.SamAccountName, p.SamAccountName);
+
+                Assert.Equal(group.Guid, p.Guid);
+                Assert.Equal(group.Sid, p.Sid);
+                Assert.Equal(group.UserPrincipalName, p.UserPrincipalName);
+                Assert.Equal(group.DistinguishedName, p.DistinguishedName);
+            }
+        }
+
+        private UserPrincipal CreateUser(PrincipalContext context, string userName, string password, string firstName, string lastName, string displayName)
+        {
+            UserPrincipal user = new UserPrincipal(context, userName, password, true);
```

Comment:
This will create the object in a well known location ( users container ) which means that concurrent execution of these tests will cause failures.  We should generate a random number during each test execution and append this to the object names that way there won't be any collisions

Reply:
The only concern with that is if some reason we created a user with random name and then the test crashes (or terminated up normally), we'll not be able to clean it up. do you think this is ok?

Reply:
I thought about this more and I agree that it's more important for us to keep a clean test environment than to support concurrent execution.  We can always add that in the future if required.  Keep the code as is.